src/algebra/star/algebra.lean

```diff
@@ -32,6 +35,22 @@ class star_algebra (R : Type u) (A : Type v)
   star (r • a) = star r • star a :=
 star_algebra.star_smul r a
 
+instance {R₁ R₂ : Type*} [semiring R₁] [semiring R₂] (e : R₁ ≃+* R₂) :
+  ring_hom_inv_pair (↑e : R₁ →+* R₂) ↑e.symm :=
+⟨e.symm_to_ring_hom_comp_to_ring_hom, e.symm.symm_to_ring_hom_comp_to_ring_hom⟩
+
+/-- If `A` is a left- and right- algebra over `R` with compatible algebra maps, then `star` is a
+semilinear equivalence. -/
+@[simps]
+def star_linear_equiv {R : Type u} {A : Type v}
+  [comm_semiring R] [star_ring R] [semiring A] [star_ring A]
+  [algebra R A] [star_algebra R A]
+  [algebra Rᵒᵖ A] [star_algebra Rᵒᵖ A] (h : ∀ (r : R) (a : A), r • a = op r • a) :
```

Comment:
We might want a typeclass for `h` here. The problem is that `has_scalar R A` and `has_scalar Rᵒᵖ A` are likely not defeq, but `algebra_map R A` and `algebra_map Rᵒᵖ A` almost certainly are. As a result, we can't provide a `algebra Rᵒᵖ A` instance from a `algebra R A` instance as doing so would create a diamond in `smul`.

Reply:
What about `linear_equiv.star` or `starₛₗ`?

Reply:
I picked this name because it matches `star_ring_equiv` and `star_add_equiv`.